connection.go

```diff
@@ -162,28 +165,183 @@ func (mc *mysqlConn) Prepare(query string) (driver.Stmt, error) {
 	return stmt, err
 }
 
+// https://github.com/mysql/mysql-server/blob/mysql-5.7.5/libmysql/libmysql.c#L1150-L1156
+func (mc *mysqlConn) escapeBytes(buf, v []byte) []byte {
```

Comment:
I'd like you to inline the `escapeBytes` and `escapeString` calls in the switch statement. They are not that long and only called from there (inline only these two, all String / Byte methods would make the switch too long for my taste).

LGTM when that's done.

Thanks @methane!